core/comm/client.go

```diff
@@ -70,6 +70,7 @@ func (client *GRPCClient) parseSecureOptions(opts SecureOptions) error {
 	}
 
 	client.tlsConfig = &tls.Config{
+		InsecureSkipVerify:    opts.InsecureSkipVerify,
```

Comment:
I really do not think we should allow this to be set.  Maybe we can allow the ServerName to be set (aka hostname override) but we should not allow this to be set.

Reply:
@mastersingh24 serverName can only be set with one server, but this connection will be used to communicate to multiple OSNs, correct?

The OSN will check to match the tlscert with those stored inside the channel config. Hence even with this option enabled, the communication is still safe.

Besides, this is useful when the OSN stays behind network middlebox or doing tests.

I've added explanation in the configuration file to warn not setting this unless clear to know what it means.

Reply:
We can use a similar approach to the address overrides we have in the peer:

```
        addressOverrides:
        #  - from:
        #    to:
        #    caCertsFile:
        #  - from:
        #    to:
        #    caCertsFile:
```

But I'm still struggling to understand why this should be solved programmatically and not by issuing correct certificates or making the DNS environment fit what is in the certificates.

Reply:
@yacovm As i mentioned, the issue to resolve here is that will mismatched certificate. If the certificate is matching, then no need to enable this option.

Reply:
But I am not sure this is the way to solve it. The entity that sets the endpoint and the entity that sets the certificate, is the same entity. When you create a certificate, you get to choose the SAN on it and it should match the endpoint. 

Besides, I'm telling you that we can map the hosts of the endpoint to different hosts locally like the peer does in the example above, instead of just using insecureSkipVerify